src/zerocash/PourTransaction.cpp

```diff
@@ -168,11 +154,21 @@ void PourTransaction::init(uint16_t version_num,
     convertBytesVectorToVector(c_1_new.getCoinCommitment().getCommitmentValue(), cm_new_1_bv);
     convertBytesVectorToVector(c_2_new.getCoinCommitment().getCommitmentValue(), cm_new_2_bv);
 
-    convertBytesVectorToVector(addr_1_old.getPrivateAddress().getAddressSecret(), addr_sk_old_1_bv);
-    convertBytesVectorToVector(addr_2_old.getPrivateAddress().getAddressSecret(), addr_sk_old_2_bv);
+    {
+        auto a = from_uint256(addr_1_old.getPrivateAddress().getAddressSecret());
+        auto b = from_uint256(addr_2_old.getPrivateAddress().getAddressSecret());
```

Comment:
If ``getAddressSecret`` returned a (typedef to a) ``boost::array<unsigned char, 32>`` then some of this type conversion would just go away.